### ISSUE TITLE
[unity] Enabling CanvasGroup compatible causes Additive effect not working.

### ISSUE THREAD

**Issue by birdkingz** [EsotericSoftware/spine-runtimes#1756]
My spine animation uses **Additive Blend** mode in the Spine Editor but when I export it to Unity, the additive effect isn't working correctly (I'm using **SkeletonGraphic** with **CanvasGroup compatible** enabled). 

I have to disable **CanvasGroup compatible** in order to see the additive effect but I need **CanvasGroup compatible** enabled because there's a **CanvasGroup** component to control all children's alpha values.

How can I fix this issue where I can have both **CanvasGroup compatible** enabled and **Additive Blend** working correctly?

You may download the [sample project](https://drive.google.com/file/d/1f6CBmiWzjtcmaGEl2fkfiKqRk8a4Vca8/view?usp=sharing).

Currently running on 
**_Unity 2018.4.25f1_** 
with the latest 
**_spine-unity 3.8 (updated 2020-09-03)_** package

**Comment by HaraldCsaszar** [comment]
From the changelog:

Now providing a `Canvas Group Tint Black` parameter a `SkeletonGraphic` Inspector in the `Advanced` section. Enable when using Additive blend mode at SkeletonGraphic under a CanvasGroup. Be sure to also have the parameter `CanvasGroup Compatible` enabled at the shader.

**Comment by HaraldCsaszar** [comment]
A new 3.8. unitypackage has just been release, available for download here as usual:
http://esotericsoftware.com/spine-unity-download

**Comment by birdkingz** [comment]
No, it didn't fix the issue! 
Please download the sample project that I provided and try it on your own.

I've turned on the Canvas Group Tint Black but there are no differences.

You might have misunderstood the Additive blend mode that I was talking about.
I didn't use Additive blend mode in Unity. The Additive blend mode was used in Spine Editor when creating animation. 
Then only export the whole animation to Unity. 
Please download the sample project to see for yourself.

**Comment by HaraldCsaszar** [comment]
Sorry for the misunderstanding and the late reply, thanks for the feedback.

I missed to mention that it is required to use the `Spine/SkeletonGraphic Tint Black` shader for using `Additive` blend mode at a `CanvasGroup`. We used our own test assets to reproduce the problem, because your setup scene neither contained a `CanvasGroup`, nor showed any additive effect at all (the starting animation was set to none, `orb_orb` would have shown the additive attachment). For the future, please always provide reproduction sample scenes that reproduce the problem without requiring any additional undocumented modifications.

The reason why `Spine/SkeletonGraphic Tint Black` is required is that the normal `Spine/SkeletonGraphic` shader is missing the additional shader channel where the alpha information is transported in an unmodified way. We do not want to over-complicate the standard `Spine/SkeletonGraphic` shader to require the additional shader channel setup at the Canvas component. Unfortunately Unity supplies only [Position, Color, and Uv0](https://docs.unity3d.com/ScriptReference/Canvas-additionalShaderChannels.html), with `Uv0` of type `float2` instead of `float3`, leaving no room to transport the unaltered alpha value without an additional vertex channel.

We have added this info to the shader documentation section:
http://esotericsoftware.com/spine-unity#Shaders

**Comment by birdkingz** [comment]
I think you misunderstood me again.

OK, let me rephrase.
Please ignore whatever additive blend mode that I mentioned.

Based on the sample scene that I provided. 
Once you **clicked** anywhere on the screen, it will play the animation.
Two SkeletonGraphics are using the same SkeletonDataAsset but both turned out to be very different (in terms of visually, color)

All I need is just to have the "Right Orb" look/appear exactly the same as the "Left Orb"
(with "Right Orb" remain using **CanvasGroup Compatible** in its shader setting)

**Comment by HaraldCsaszar** [comment]
It seems that my answer was not phrased clearly enough:

It is required to use the `Spine/SkeletonGraphic Tint Black` shader for using `Additive` blend mode when having `CanvasGroup Compatible` enabled (which is mandatory for proper blending when using a `CanvasGroup`).
Please read the above answer again for more details about why this is necessary.

Is this more clear now?

**Comment by HaraldCsaszar** [comment]
FYI: We have recently fixed related blending issues (in the above commit 5c1faa8 ), some problems were introduced by recent changes.

New unitypackages are available for download here as usual:
http://esotericsoftware.com/spine-unity-download

**Comment by birdkingz** [comment]
> FYI: We have recently fixed related blending issues (in the above commit [5c1faa8](https://github.com/EsotericSoftware/spine-runtimes/commit/5c1faa85af9ed7a7d265bb8da3c0a5235454fa59) ), some problems were introduced by recent changes.
> 
> New unitypackages are available for download here as usual:
> http://esotericsoftware.com/spine-unity-download

Finally, this fixes the issue that I encountered.

PS: Once again I would like to clarify I do not want to have additive effect.
Changing the shader to use `Spine/SkeletonGraphic Tint Black` is not what I'm looking for.

**Comment by HaraldCsaszar** [comment]
Glad it fixed your issue.

PS: I always assumed that you want the Spine slot blend mode `Additive` that you authored on the provided sample asset (slots `orb_fx_glow` and `orb_fx_round` for example) to behave the same in Unity as in Spine. Your statement "Once again I would like to clarify I do not want to have additive effect." makes little sense in the context of the above postings of yours and is rather confusing. I assume that you do want your additive slots to blend in an additive way. I never said that you should change the shader to `Spine/SkeletonGraphic Tint Black` to use the Tint Black feature, but to do this as it is necessary to resolve this problem of yours: "How can I fix this issue where I can have both CanvasGroup compatible enabled and Additive Blend [interpreted as slot blend mode Additive] working correctly?". I am writing this in case that you or anyone else encounters a similar problem in the future.
